.github/workflows/r-unit-tests.yaml

```diff
@@ -0,0 +1,25 @@
+name: Run R Unit Tests
+
+on: [push]
+
+jobs:
+  run-tests:
+    name: R Unit Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - name: Install dependencies
+        uses: r-lib/actions/setup-r-dependencies@v2
+
+      - name: Run unit tests
+        run: |
+          testthat::test_dir("tests/testthat/")
```

Comment:
Can you explain why you want this vs just doing all the testing in r check and code coverage as people normally do?

Reply:
I like having the tests run post-push on all branches. R CMD check/codecov are set to PR-only.

Reply:
maybe we should make them run on PR draft rather than post-push?

Reply:
So they will run on PR draft. I think its fine if you want them for all pushes but just make it bee Rmd check and  control wastage with concurrency groups? 

Personally i just make draft Prs to get this and I like that as it encourages me to do so early on if I want CI